lib/src/logger.dart

```diff
@@ -95,19 +103,19 @@ class Logger {
     if (parent == null) {
       // We're either the root logger or a detached logger.  Return our own
       // level.
-      effectiveLevel = _level;
+      effectiveLevel = _level!;
     } else if (!hierarchicalLoggingEnabled) {
-      effectiveLevel = root._level;
+      effectiveLevel = root._level!;
     } else {
-      effectiveLevel = _level ?? parent.level;
+      effectiveLevel = _level ?? parent!.level;
     }
 
-    assert(effectiveLevel != null);
+    assert((effectiveLevel as dynamic) != null);
```

Comment:
@stereotype441 - I think we still want this check in weak mode.

Should we look for a more ergonomic way to allow this without the `unnecessary_null_comparison` warning?

@jakemac53 - I do wonder whether using `// ignore: unnecessary_null_comparison` would be better here in the long term since it's easier to grep for once we no longer need to support weak mode?

Reply:
I opted for the ignore, that seems better